### ISSUE TITLE
Execution failed for task ':core:invoker:compileScala'.

### ISSUE THREAD

**Issue by GitHubDiom** [apache/openwhisk#5124]
## Environment details:

* vagrant: ubuntu/binoic64, box version: 20210415.0.0
* Docker version 19.03.5, build 633a0ea838
* Run script `tools/ubuntu-setup/all.sh` with `sudo` to install the dependencies

## Steps to reproduce the issue:

1.   ./gradlew core:standalone:bootRun
2.   
3.   

## Provide the expected results and outputs:

```
The OpenWhisk stack should up
```

## Provide the actual results and outputs:

```
vagrant@wsk-01:~/openwhisk$ ./gradlew core:standalone:bootRun --stacktrace
Build using Scala 2.12

> Configure project :core:scheduler
[task ':core:scheduler:assemble', task ':core:scheduler:assembleDist', task ':core:scheduler:build', task ':core:scheduler:buildDependents', task ':core:scheduler:buildEnvironment', task ':core:scheduler:buildNeeded', task ':core:scheduler:check', task ':core:scheduler:checkScalafmt', task ':core:scheduler:checkScalafmtAll', task ':core:scheduler:checkScoverage', task ':core:scheduler:checkScoverageScalafmt', task ':core:scheduler:checkTestScalafmt', task ':core:scheduler:classes', task ':core:scheduler:clean', task ':core:scheduler:cleanEclipse', task ':core:scheduler:cleanEclipseClasspath', task ':core:scheduler:cleanEclipseJdt', task ':core:scheduler:cleanEclipseProject', task ':core:scheduler:compileJava', task ':core:scheduler:compileScala', task ':core:scheduler:compileScoverageJava', task ':core:scheduler:compileScoverageScala', task ':core:scheduler:compileTestJava', task ':core:scheduler:compileTestScala', task ':core:scheduler:components', task ':core:scheduler:dependencies', task ':core:scheduler:dependencyInsight', task ':core:scheduler:dependentComponents', task ':core:scheduler:distTar', task ':core:scheduler:distZip', task ':core:scheduler:eclipse', task ':core:scheduler:eclipseClasspath', task ':core:scheduler:eclipseJdt', task ':core:scheduler:eclipseProject', task ':core:scheduler:help', task ':core:scheduler:install', task ':core:scheduler:installDist', task ':core:scheduler:jar', task ':core:scheduler:javadoc', task ':core:scheduler:model', task ':core:scheduler:prepareKotlinBuildScriptModel', task ':core:scheduler:processResources', task ':core:scheduler:processScoverageResources', task ':core:scheduler:processTestResources', task ':core:scheduler:projects', task ':core:scheduler:properties', task ':core:scheduler:reportScoverage', task ':core:scheduler:run', task ':core:scheduler:scaladoc', task ':core:scheduler:scalafmt', task ':core:scheduler:scalafmtAll', task ':core:scheduler:scoverageClasses', task ':core:scheduler:scoverageScalafmt', task ':core:scheduler:startScripts', task ':core:scheduler:tasks', task ':core:scheduler:test', task ':core:scheduler:testClasses', task ':core:scheduler:testScalafmt']
Scala sub-project 'gatling_tests' doesn't have Scoverage applied and will be ignored in parent project aggregation

> Task :core:invoker:compileScala
Pruning sources from previous analysis, due to incompatible CompileSetup.
/home/vagrant/openwhisk/core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/v2/ActivationClientProxy.scala:76: class ActorMaterializer in package stream is deprecated (since 2.6.0): The Materializer now has all methods the ActorMaterializer used to have
  mat: ActorMaterializer,
       ^
/home/vagrant/openwhisk/core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/v2/ActivationClientProxy.scala:414: class ActorMaterializer in package stream is deprecated (since 2.6.0): The Materializer now has all methods the ActorMaterializer used to have
                                                                                   mat: ActorMaterializer,
                                                                                        ^
/home/vagrant/openwhisk/core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/v2/FunctionPullingContainerProxy.scala:197: class ActorMaterializer in package stream is deprecated (since 2.6.0): The Materializer now has all methods the ActorMaterializer used to have
  testTcp: Option[ActorRef])(implicit actorSystem: ActorSystem, mat: ActorMaterializer, logging: Logging)
                                                                     ^
/home/vagrant/openwhisk/core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/v2/FunctionPullingContainerProxy.scala:376: method setTimer in trait FSM is deprecated (since 2.6.0): Use startSingleTimer, startTimerWithFixedDelay or startTimerAtFixedRate instead. This has the same semantics as startTimerAtFixedRate, but startTimerWithFixedDelay is often preferred.
      setTimer(UnusedTimeoutName, StateTimeout, unusedTimeout)
      ^
/home/vagrant/openwhisk/core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/v2/FunctionPullingContainerProxy.scala:509: method setTimer in trait FSM is deprecated (since 2.6.0): Use startSingleTimer, startTimerWithFixedDelay or startTimerAtFixedRate instead. This has the same semantics as startTimerAtFixedRate, but startTimerWithFixedDelay is often preferred.
      setTimer(UnusedTimeoutName, StateTimeout, unusedTimeout)
      ^
/home/vagrant/openwhisk/core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/v2/FunctionPullingContainerProxy.scala:616: method setTimer in trait FSM is deprecated (since 2.6.0): Use startSingleTimer, startTimerWithFixedDelay or startTimerAtFixedRate instead. This has the same semantics as startTimerAtFixedRate, but startTimerWithFixedDelay is often preferred.
        setTimer(RunningActivationTimeoutName, ClientClosed, runningActivationTimeout)
        ^
/home/vagrant/openwhisk/core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/v2/FunctionPullingContainerProxy.scala:677: method setTimer in trait FSM is deprecated (since 2.6.0): Use startSingleTimer, startTimerWithFixedDelay or startTimerAtFixedRate instead. This has the same semantics as startTimerAtFixedRate, but startTimerWithFixedDelay is often preferred.
          setTimer(UnusedTimeoutName, StateTimeout, unusedTimeout)
          ^
/home/vagrant/openwhisk/core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/v2/FunctionPullingContainerProxy.scala:718: method setTimer in trait FSM is deprecated (since 2.6.0): Use startSingleTimer, startTimerWithFixedDelay or startTimerAtFixedRate instead. This has the same semantics as startTimerAtFixedRate, but startTimerWithFixedDelay is often preferred.
      setTimer(KeepingTimeoutName, Remove, warmedContainerKeepingTimeout)
      ^
/home/vagrant/openwhisk/core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/v2/FunctionPullingContainerProxy.scala:780: method setTimer in trait FSM is deprecated (since 2.6.0): Use startSingleTimer, startTimerWithFixedDelay or startTimerAtFixedRate instead. This has the same semantics as startTimerAtFixedRate, but startTimerWithFixedDelay is often preferred.
    case _ -> Paused   => setTimer(IdleTimeoutName, StateTimeout, idleTimeout)
                          ^
/home/vagrant/openwhisk/core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/v2/FunctionPullingContainerProxy.scala:1248: class ActorMaterializer in package stream is deprecated (since 2.6.0): The Materializer now has all methods the ActorMaterializer used to have
            tcp: Option[ActorRef] = None)(implicit actorSystem: ActorSystem, mat: ActorMaterializer, logging: Logging) =
                                                                                  ^
10 errors found

> Task :core:invoker:compileScala FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':core:invoker:compileScala'.
> Compilation failed

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':core:invoker:compileScala'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.accept(ExecuteActionsTaskExecuter.java:148)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.accept(ExecuteActionsTaskExecuter.java:145)
        at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:191)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:138)
        at org.gradle.api.internal.tasks.execution.ResolveBeforeExecutionStateTaskExecuter.execute(ResolveBeforeExecutionStateTaskExecuter.java:75)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:108)
        at org.gradle.api.internal.tasks.execution.ResolveBeforeExecutionOutputsTaskExecuter.execute(ResolveBeforeExecutionOutputsTaskExecuter.java:67)
        at org.gradle.api.internal.tasks.execution.StartSnapshotTaskInputsBuildOperationTaskExecuter.execute(StartSnapshotTaskInputsBuildOperationTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.ResolveAfterPreviousExecutionStateTaskExecuter.execute(ResolveAfterPreviousExecutionStateTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:94)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:95)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:73)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:49)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:49)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:43)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:355)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:343)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:336)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:322)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$1.execute(DefaultPlanExecutor.java:134)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$1.execute(DefaultPlanExecutor.java:129)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:202)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:193)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:129)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.api.internal.tasks.compile.CompilationFailedException: Compilation failed
        at org.gradle.api.internal.tasks.scala.ZincScalaCompiler$Compiler.execute(ZincScalaCompiler.java:88)
        at org.gradle.api.internal.tasks.scala.ZincScalaCompiler.execute(ZincScalaCompiler.java:57)
        at org.gradle.api.internal.tasks.scala.ZincScalaCompiler.execute(ZincScalaCompiler.java:40)
        at org.gradle.api.internal.tasks.compile.daemon.AbstractDaemonCompiler$CompilerCallable.call(AbstractDaemonCompiler.java:99)
        at org.gradle.api.internal.tasks.compile.daemon.AbstractDaemonCompiler$CompilerCallable.call(AbstractDaemonCompiler.java:79)
        at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:44)
        at org.gradle.workers.internal.AbstractClassLoaderWorker.executeInClassLoader(AbstractClassLoaderWorker.java:34)
        at org.gradle.workers.internal.IsolatedClassloaderWorker.execute(IsolatedClassloaderWorker.java:55)
        at org.gradle.workers.internal.WorkerDaemonServer.execute(WorkerDaemonServer.java:55)
        at org.gradle.process.internal.worker.request.WorkerAction.run(WorkerAction.java:116)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
        at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:175)
        at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:157)
        at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
        ... 3 more
Caused by: Compilation failed
        at sbt.compiler.AnalyzingCompiler.call(AnalyzingCompiler.scala:111)
        at sbt.compiler.AnalyzingCompiler.compile(AnalyzingCompiler.scala:53)
        at sbt.compiler.AnalyzingCompiler.compile(AnalyzingCompiler.scala:47)
        at sbt.compiler.AggressiveCompile$$anonfun$3$$anonfun$compileScala$1$1.apply$mcV$sp(AggressiveCompile.scala:95)
        at sbt.compiler.AggressiveCompile$$anonfun$3$$anonfun$compileScala$1$1.apply(AggressiveCompile.scala:95)
        at sbt.compiler.AggressiveCompile$$anonfun$3$$anonfun$compileScala$1$1.apply(AggressiveCompile.scala:95)
        at sbt.compiler.AggressiveCompile.sbt$compiler$AggressiveCompile$$timed(AggressiveCompile.scala:168)
        at sbt.compiler.AggressiveCompile$$anonfun$3.compileScala$1(AggressiveCompile.scala:94)
        at sbt.compiler.AggressiveCompile$$anonfun$3.apply(AggressiveCompile.scala:142)
        at sbt.compiler.AggressiveCompile$$anonfun$3.apply(AggressiveCompile.scala:84)
        at sbt.inc.IncrementalCompile$$anonfun$doCompile$1.apply(Compile.scala:66)
        at sbt.inc.IncrementalCompile$$anonfun$doCompile$1.apply(Compile.scala:64)
        at sbt.inc.IncrementalCommon.cycle(IncrementalCommon.scala:32)
        at sbt.inc.Incremental$$anonfun$1.apply(Incremental.scala:72)
        at sbt.inc.Incremental$$anonfun$1.apply(Incremental.scala:71)
        at sbt.inc.Incremental$.manageClassfiles(Incremental.scala:99)
        at sbt.inc.Incremental$.compile(Incremental.scala:71)
        at sbt.inc.IncrementalCompile$.apply(Compile.scala:54)
        at sbt.compiler.AggressiveCompile.compile2(AggressiveCompile.scala:159)
        at sbt.compiler.AggressiveCompile.compile1(AggressiveCompile.scala:68)
        at com.typesafe.zinc.Compiler.compile(Compiler.scala:207)
        at com.typesafe.zinc.Compiler.compile(Compiler.scala:189)
        at com.typesafe.zinc.Compiler.compile(Compiler.scala:180)
        at com.typesafe.zinc.Compiler.compile(Compiler.scala:171)
        at org.gradle.api.internal.tasks.scala.ZincScalaCompiler$Compiler.execute(ZincScalaCompiler.java:86)
        ... 17 more


* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.5.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 25s
12 actionable tasks: 2 executed, 10 up-to-date
```

## Additional information you deem important:
* issue happens always
* changes nothing

**Comment by jiangpengcheng** [comment]
sorry for the error, I think it should be fixed by this https://github.com/apache/openwhisk/pull/5123

**Comment by GitHubDiom** [comment]
> sorry for the error, I think it should be fixed by this #5123

Thanks for the reply, but another problem seems to have cropped up

```shell
vagrant@wsk-01:~/openwhisk$ ./gradlew core:standalone:bootRun
Build using Scala 2.12

> Configure project :core:scheduler
[task ':core:scheduler:assemble', task ':core:scheduler:assembleDist', task ':core:scheduler:build', task ':core:scheduler:buildDependents', task ':core:scheduler:buildEnvironment', task ':core:scheduler:buildNeeded', task ':core:scheduler:check', task ':core:scheduler:checkScalafmt', task ':core:scheduler:checkScalafmtAll', task ':core:scheduler:checkScoverage', task ':core:scheduler:checkScoverageScalafmt', task ':core:scheduler:checkTestScalafmt', task ':core:scheduler:classes', task ':core:scheduler:clean', task ':core:scheduler:cleanEclipse', task ':core:scheduler:cleanEclipseClasspath', task ':core:scheduler:cleanEclipseJdt', task ':core:scheduler:cleanEclipseProject', task ':core:scheduler:compileJava', task ':core:scheduler:compileScala', task ':core:scheduler:compileScoverageJava', task ':core:scheduler:compileScoverageScala', task ':core:scheduler:compileTestJava', task ':core:scheduler:compileTestScala', task ':core:scheduler:components', task ':core:scheduler:dependencies', task ':core:scheduler:dependencyInsight', task ':core:scheduler:dependentComponents', task ':core:scheduler:distTar', task ':core:scheduler:distZip', task ':core:scheduler:eclipse', task ':core:scheduler:eclipseClasspath', task ':core:scheduler:eclipseJdt', task ':core:scheduler:eclipseProject', task ':core:scheduler:help', task ':core:scheduler:install', task ':core:scheduler:installDist', task ':core:scheduler:jar', task ':core:scheduler:javadoc', task ':core:scheduler:model', task ':core:scheduler:prepareKotlinBuildScriptModel', task ':core:scheduler:processResources', task ':core:scheduler:processScoverageResources', task ':core:scheduler:processTestResources', task ':core:scheduler:projects', task ':core:scheduler:properties', task ':core:scheduler:reportScoverage', task ':core:scheduler:run', task ':core:scheduler:scaladoc', task ':core:scheduler:scalafmt', task ':core:scheduler:scalafmtAll', task ':core:scheduler:scoverageClasses', task ':core:scheduler:scoverageScalafmt', task ':core:scheduler:startScripts', task ':core:scheduler:tasks', task ':core:scheduler:test', task ':core:scheduler:testClasses', task ':core:scheduler:testScalafmt']
Scala sub-project 'gatling_tests' doesn't have Scoverage applied and will be ignored in parent project aggregation

> Task :core:standalone:copyGWActions FAILED

FAILURE: Build failed with an exception.

* Where:
Build file '/home/vagrant/openwhisk/core/standalone/build.gradle' line: 64

* What went wrong:
Execution failed for task ':core:standalone:copyGWActions'.
> Execute failed: java.io.IOException: Cannot run program "npm" (in directory "/home/vagrant/openwhisk/core/routemgmt/createApi"): error=2, No such file or directory

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.5.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 1s
23 actionable tasks: 3 executed, 20 up-to-date
```

**Comment by GitHubDiom** [comment]
> > sorry for the error, I think it should be fixed by this #5123
> 
> Thanks for the reply, but another problem seems to have cropped up
> 
> ```shell
> vagrant@wsk-01:~/openwhisk$ ./gradlew core:standalone:bootRun
> Build using Scala 2.12
> 
> > Configure project :core:scheduler
> [task ':core:scheduler:assemble', task ':core:scheduler:assembleDist', task ':core:scheduler:build', task ':core:scheduler:buildDependents', task ':core:scheduler:buildEnvironment', task ':core:scheduler:buildNeeded', task ':core:scheduler:check', task ':core:scheduler:checkScalafmt', task ':core:scheduler:checkScalafmtAll', task ':core:scheduler:checkScoverage', task ':core:scheduler:checkScoverageScalafmt', task ':core:scheduler:checkTestScalafmt', task ':core:scheduler:classes', task ':core:scheduler:clean', task ':core:scheduler:cleanEclipse', task ':core:scheduler:cleanEclipseClasspath', task ':core:scheduler:cleanEclipseJdt', task ':core:scheduler:cleanEclipseProject', task ':core:scheduler:compileJava', task ':core:scheduler:compileScala', task ':core:scheduler:compileScoverageJava', task ':core:scheduler:compileScoverageScala', task ':core:scheduler:compileTestJava', task ':core:scheduler:compileTestScala', task ':core:scheduler:components', task ':core:scheduler:dependencies', task ':core:scheduler:dependencyInsight', task ':core:scheduler:dependentComponents', task ':core:scheduler:distTar', task ':core:scheduler:distZip', task ':core:scheduler:eclipse', task ':core:scheduler:eclipseClasspath', task ':core:scheduler:eclipseJdt', task ':core:scheduler:eclipseProject', task ':core:scheduler:help', task ':core:scheduler:install', task ':core:scheduler:installDist', task ':core:scheduler:jar', task ':core:scheduler:javadoc', task ':core:scheduler:model', task ':core:scheduler:prepareKotlinBuildScriptModel', task ':core:scheduler:processResources', task ':core:scheduler:processScoverageResources', task ':core:scheduler:processTestResources', task ':core:scheduler:projects', task ':core:scheduler:properties', task ':core:scheduler:reportScoverage', task ':core:scheduler:run', task ':core:scheduler:scaladoc', task ':core:scheduler:scalafmt', task ':core:scheduler:scalafmtAll', task ':core:scheduler:scoverageClasses', task ':core:scheduler:scoverageScalafmt', task ':core:scheduler:startScripts', task ':core:scheduler:tasks', task ':core:scheduler:test', task ':core:scheduler:testClasses', task ':core:scheduler:testScalafmt']
> Scala sub-project 'gatling_tests' doesn't have Scoverage applied and will be ignored in parent project aggregation
> 
> > Task :core:standalone:copyGWActions FAILED
> 
> FAILURE: Build failed with an exception.
> 
> * Where:
> Build file '/home/vagrant/openwhisk/core/standalone/build.gradle' line: 64
> 
> * What went wrong:
> Execution failed for task ':core:standalone:copyGWActions'.
> > Execute failed: java.io.IOException: Cannot run program "npm" (in directory "/home/vagrant/openwhisk/core/routemgmt/createApi"): error=2, No such file or directory
> 
> * Try:
> Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> 
> * Get more help at https://help.gradle.org
> 
> Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
> Use '--warning-mode all' to show the individual deprecation warnings.
> See https://docs.gradle.org/5.5.1/userguide/command_line_interface.html#sec:command_line_warnings
> 
> BUILD FAILED in 1s
> 23 actionable tasks: 3 executed, 20 up-to-date
> ```

solved by `sudo apt install npm`
